tapir/coop/tests/membership_resignation/test_delete_view.py

```diff
@@ -69,6 +69,7 @@ def test_membershipResignationDeleteView_default_sharesEndDateSetToNone(
 
         self.assertStatusCode(response, HTTPStatus.OK)
         mock_on_resignation_deleted.assert_called_once_with(resignation)
+        self.assertTrue(resignation.share_owner.share_ownerships.first().end_date, None)
```

Comment:
I think this should be `self.assertEqual` instead of `self.assertTrue`

It's a good idea to update the test, but this is not the correct test to update: this one only checks that `on_resignation_deleted` gets called. Since `on_resignation_deleted` is replaced with a mock function, it is not actually executed, inside the test the dates don't actually get updated.

I think we want a new test, similar to `test_deleteEndDates_default_sharesEndDateSetToNone`, that instead does `test_deleteEndDates_sharesEndOnPayOutDay_sharesEndDateSetToNone`

Reply:
I think this is done by `test_deleteEndDates_someSharesEndedBeforeCancellation_thoseSharesKeepTheirEndDate` in `test_service.py` in the last line. Don't you think?

Reply:
Not quite: in `test_deleteEndDates_someSharesEndedBeforeCancellation_thoseSharesKeepTheirEndDate`, we check that shares that ended before the cancellation are not affected. 

In case of share buy back, we want to check that shares that end after the pay out day have their end-date set to none, and that shares that end before the pay out day are not affected.